returnn/util/file_cache.py

```diff
@@ -282,24 +290,24 @@ def _copy_file_if_needed(self, src_filename: str, dst_filename: str):
         """
         Copy the file to the cache directory.
         """
-        if self._check_existing_copied_file_maybe_cleanup(src_filename, dst_filename):
-            os.utime(dst_filename, None)  # touch
-            return
-
         # Create dirs.
         dst_dir = os.path.dirname(dst_filename)
         os.makedirs(dst_dir, exist_ok=True)
 
         # Copy the file, while holding a lock. See comment on lock_timeout above.
-        with LockFile(directory=self.cache_directory, name="dir.lock", lock_timeout=self._lock_timeout) as lock:
+        with LockFile(
+            directory=dst_dir, name=os.path.basename(dst_filename) + ".lock", lock_timeout=self._lock_timeout
+        ) as lock:
             # Maybe it was copied in the meantime, while waiting for the lock.
             if self._check_existing_copied_file_maybe_cleanup(src_filename, dst_filename):
+                print(f"FileCache: using existing file {dst_filename}")
+                os.utime(dst_filename, None)  # touch
```

Comment:
I was not doing this touch in my earlier code because I thought when some other proc has just copied it (before the lock, it was not there...), it should already be up-to-date. But it probably doesn't matter.

Reply:
exactly